core/src/test/java/org/elasticsearch/index/query/InnerHitBuilderTests.java

```diff
@@ -221,7 +221,12 @@ public static InnerHitBuilder randomInnerHits(boolean recursive, boolean include
         innerHits.setTrackScores(randomBoolean());
         innerHits.setFieldNames(randomListStuff(16, () -> randomAsciiOfLengthBetween(1, 16)));
         innerHits.setFieldDataFields(randomListStuff(16, () -> randomAsciiOfLengthBetween(1, 16)));
-        innerHits.setScriptFields(new HashSet<>(randomListStuff(16, InnerHitBuilderTests::randomScript)));
+        // Random script fields deduped on their field name.
+        Map<String, SearchSourceBuilder.ScriptField> scriptFields = new HashMap<>();
```

Comment:
I think that is pretty much the same thing it used to be? ScriptField's hashCode and equals care about all three of it's fields but the round trip won't work if the field names aren't all unique.